### PR TITLE
Added gitlab_runner module

### PR DESCRIPTION
**Pull request by SamyCoenen**
##### SUMMARY

Added the gitlab_runner module to create, update and delete a Gitlab Runner. This helps with creating idempotent playbooks and roles for Gitlab Runner installations.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

gitlab_runner

##### ANSIBLE VERSION

```
ansible 2.7.1
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/samy/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.7.1 (default, Oct 22 2018, 10:41:28) [GCC 8.2.1 20180831]
```

##### ADDITIONAL INFORMATION

This module uses the Gitlab v4 API, there are no external dependencies.

### CODE REVIEW

**Comment by marwatk** [comment]
shipit

**Comment by bincyber** [comment]
shipit

File: lib/ansible/modules/source_control/gitlab_runner.py [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: gitlab_runner
+short_description: Create, modify and delete Gitlab Runners.
```
Comment (adrianmoisey) [inline lib/ansible/modules/source_control/gitlab_runner.py]: All occurrences of 'Gitlab' should be 'GitLab"

File: lib/ansible/modules/source_control/gitlab_runner.py [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+description:
+  - Register, update and delete runners with the Gitlab API.
+  - All operations are performed using the Gitlab API v4.
+  - For details, consult the full API documentation at U(https://docs.gitlab.com/ee/api/runners.html).
+  - A valid private API token is required for all operations. You can create as many tokens as you like using the Gitlab webinterface at
```
Comment (adrianmoisey) [inline lib/ansible/modules/source_control/gitlab_runner.py]: "webinterface" should be "web interface"

**Comment by gundalow** [comment]
@SamyCoenen Thank you for your PR
@marwatk @adrianmoisey Thank you for the reviews

I've pushed two commits directly to this PR to address the remaining issues from @adrianmoisey 

rebuild_merge

**Comment by adrianmoisey** [comment]
LGTM

**Comment by adrianmoisey** [comment]
shipit

**Comment by gundalow** [comment]
Thank you all. Merged into `devel` so this module will be available in Ansible 2.8

Having a look at https://ansible.sivel.net/pr/byfile.html I see some other GitLab PRs that could benefit from your reviews:

* Make use of a better python module for gitlab_project #40767
* New module: Gitlab service #40053
* New module: Gitlab webhook #39991

**Pull request merged**